README.md

```diff
@@ -133,30 +134,125 @@ The relayer will now watch the contract on Ropsten and create a claim whenever i
- _token = [DEPLOYED_TOKEN_ADDRESS] *(erc20 not currently supported, enter "0x0000000000000000000000000000000000000000" for ethereum)*
- _amount = [WEI_AMOUNT]
+$ cd testnet-contracts/
 ```
 
-- 7.  Enter the same number from \_amount as the transaction's value (in wei)
-- 8.  Select "transact" to send the lock() transaction
+Start a local blockchain
 
-Then, wait for the transaction to confirm and mine, and for the relayer to pick it up. You should see the successful output in the relayer console. You can also confirm the tokens have been minted by using the CLI again:
+```
+$ yarn develop
+```
+
+### Terminal 2: Compile, deploy, check contract's deployed address
+
+Navigate to directory
+
+```
+$ cd testnet-contracts/
+```
+
+Copy contract ABI to go modules:
+
+```
+$ yarn peggy:abi
+```
+
+Deploy contract to local blockchain:
+
+```
+$ yarn migrate
+```
+
+View locally deployed peggy's address (will be logged to console):
+
+```
+$ yarn peggy:address
+```
+
+### Terminal 3: Build Ethereum Bridge and start
+
+Build the Ethereum Bridge application
+
+```
+$ make install
+```
+
+Start the Bridge's blockchain
+
+```
+$ ebd start
+```
+
+### Terminal 4: Begin the relayer service
+
+Start ebrelayer on a local web socket with peggy's deployed address from the earlier step.
+
+- Example [LOCAL_WEB_SOCKET] : ws://127.0.0.1:8545/
+- Example [PEGGY_DEPLOYED_ADDRESS] = 0xC4cE93a5699c68241fc2fB503Fb0f21724A624BB
+
+```
+$ ebrelayer init [LOCAL_WEB_SOCKET] [PEGGY_DEPLOYED_ADDRESS] LogLock\(bytes32,address,bytes,address,uint256,uint256\) validator --chain-id=testing
+```
+
+### Using Terminal 2: Send lock transaction to peggy
+
+The lock transaction has default parameters:
+
+- [HASHED_COSMOS_RECIPIENT_ADDRESS] = 0x636f736d6f7331706a74677530766175326d35326e72796b64707a74727438383761796b756530687137646668
+- [DEPLOYED_TOKEN_ADDRESS] = 0x0000000000000000000000000000000000000000
+- [WEI_AMOUNT] = 10
+
+```
+$ yarn peggy:lock
+```
+
+Expected successful output in the relayer console:
+
+```
+New Lock Transaction:
+Tx hash: 0x83e6ee88c20178616e68fee2477d21e84f16dcf6bac892b18b52c000345864c0
+Block number: 5
+Event ID: cc10955295e555130c865949fb1fd48dba592d607ae582b43a2f3f0addce83f2
+Token: 0x0000000000000000000000000000000000000000
+Sender: 0xc230f38FF05860753840e0d7cbC66128ad308B67
+Recipient: cosmos1pjtgu0vau2m52nrykdpztrt887aykue0hq7dfh
+Value: 10
+Nonce: 1
+```
+
+You can also confirm the tokens have been minted by using the CLI again:
 
 ```
 ebcli query account cosmos1pjtgu0vau2m52nrykdpztrt887aykue0hq7dfh --trust-node
 ```
 
+### Running on the testnet
+
+To run the Ethereum Bridge on the testnet, ebrelayer must be initalized with websocket provider: wss://ropsten.infura.io/ws. You'll need to specify the ropsten network via a --network flag for the following commands:
+
```

Comment:
can we group a lot of these commands/steps into a single command block with comments where explanations are needed, so that they can be copy pasted all together easily?

Reply:
Grouped the commands and added comments inline in latest. Restructured a bit to make the steps clearer as well.